core/src/main/java/com/amplifyframework/storage/StorageCategoryBehavior.java

```diff
@@ -201,6 +202,65 @@
         @NonNull Consumer<StorageUploadFileResult> onSuccess,
         @NonNull Consumer<StorageException> onError);
 
+    /**
+     * Upload a local InputStream, storing it as a remote resource.
+     * Register consumers to obtain the results of the upload.
+     * @param key the unique identifier of the object in storage
+     * @param local the local inputStream
+     * @param onSuccess Called if operation completed successfully and furnishes a result
+     * @param onError Called if an error occurs during operation
+     * @return an operation object that provides notifications and
+     *         actions related to the execution of the work
+     */
+    @NonNull
+    StorageUploadFileOperation<?> uploadInputStream(
+        @NonNull String key,
+        @NonNull InputStream local,
+        @NonNull Consumer<StorageUploadFileResult> onSuccess,
+        @NonNull Consumer<StorageException> onError);
+
+    /**
+     * Upload a local InputStream, storing it as a remote resource.
+     * Specify options such as the access level the file should have.
+     * Register consumers to observe results of upload request,
+     * as well as intermediary progress updates.
+     * @param key the unique identifier of the object in storage
+     * @param local the local inputStream
+     * @param options parameters specific to plugin behavior
+     * @param onSuccess Called if operation completed successfully and furnishes a result
+     * @param onError Called if an error occurs during operation
+     * @return an operation object that provides notifications and
+     *         actions related to the execution of the work
+     */
+    StorageUploadFileOperation<?> uploadInputStream(
+        @NonNull String key,
+        @NonNull InputStream local,
+        @NonNull StorageUploadFileOptions options,
```

Comment:
Could we create a `StorageUploadInputStreamOptions` class to use here instead?   For now, they will have the exact same set of options, but a distinct class will allow them to diverge in the future if needed.

To avoid code duplication, you could create a common superclass `StorageUploadOptions`, which is extended by `StorageUploadFileOptions` and `StorageUploadInputStreamOptions`, like this:

```
public class StorageUploadOptions extends StorageOptions {
    private final String contentType;
    private final Map<String, String> metadata;
}

public class StorageUploadFileOptions extends StorageUploadOptions { }
public class StorageUploadInputStreamOptions extends StorageUploadOptions { }
```